### DISCUSSION TITLE
Tekton Persistent volumes

### DISCUSSION THREAD

**Discussion by hiranlakshika** [tektoncd/community#372] [Q&A] [answered]
I'm really new to Tekton. According to the documentation, we need to provide a Persistent Volume for storage purposes. What is stored in the persistent volumes required by Tekton?

**Comment by ghost** [comment, marked as answer]
Hey @hiranlakshika persistent storage is used to pass data between tasks in a pipeline.  if task A checks out code from git and task B tests it then some persistent storage is required to get the code from task A's pod over to task B's pod.  This happens through either the deprecated PipelineResources mechanism ([documented here](https://github.com/tektoncd/pipeline/blob/master/docs/install.md#configuring-pipelineresource-storage)) or via [workspaces](https://github.com/tektoncd/pipeline/blob/master/docs/workspaces.md#using-persistentvolumeclaims-as-volumesource).

For anyone new to Tekton I'd generally advise against using PipelineResources if possible.
